src/app/pages/game/game.component.ts

```diff
@@ -97,6 +99,8 @@ export class GameComponent implements OnInit {
 
   ngOnInit(): void {
     this.gameService.loadData().then(() => this.onResize());
+    setInterval(() => this.checkLevelIncrease(), 3000);
```

Comment:
неплохая реализация через интервал, но у тебя `session.levelSignal` (который ты получаешь в методе `levelRank`) является сигналом, и имеет возможность говорить, когда он получил изменения
например, это можно сделать так:
```suggestion
    toObservable(this.session.levelSignal).subscribe((level) =>{
    	this.checkLevelIncrease();
    });
```
почитай как лучше считывать изменения от WritableSignal в Angular, это лишь один из примеров